request_test.go

```diff
@@ -1601,3 +1603,85 @@ func TestRequestErrorConflictMultipart(t *testing.T) {
 	req3.WithFileBytes("a", "a", []byte("a"))
 	req3.chain.assertFailed(t)
 }
+
+func TestRequestRedirect(t *testing.T) {
+	reporter := newMockReporter(t)
+
+	t.Run("dont follow redirects policy", func(t *testing.T) {
+		client := &http.Client{
+			Transport: mockTransportWithNoRedirect{t: t},
+		}
+
+		config := Config{
+			Client:   client,
+			Reporter: reporter,
+		}
+
+		req := NewRequest(config, "METHOD", "url").
+			WithHeader("Location", "/redirect").
```

Comment:
Is this header really needed? This header is usually present in responses, not in requests. On the other hand, I suggest to set Location in response, because http.Client probably expects it.

Same for the second test.

Reply:
previously i got error when not setting the `Location` header in request

but let me double check on that

Reply:
i think i read the documentation wrong

yep, you're right, the Location header should be in response not request

```
loc := resp.Header.Get("Location")
			if loc == "" {
				// While most 3xx responses include a Location, it is not
				// required and 3xx responses without a Location have been
				// observed in the wild. See issues #17773 and #49281.
				return resp, nil
			}
```

https://cs.opensource.google/go/go/+/refs/tags/go1.19.3:src/net/http/client.go;l=631;drc=d5de62df152baf4de6e9fe81933319b86fd95ae4